internal/server/routes.go

```diff
@@ -112,6 +126,10 @@ func (a *API) registerRoutes(router *gin.RouterGroup, promRegistry prometheus.Re
 	v1.GET("/machines/:id/grants", removed("v0.9.0"))
 }
 
+type ReqHandlerFunc[Req any] func(c *gin.Context, req *Req) error
+type ResHandlerFunc[Res any] func(c *gin.Context) (Res, error)
+type ReqResHandlerFunc[Req, Res any] func(c *gin.Context, req *Req) (Res, error)
```

Comment:
I moved these down to just above the first place they are referenced.

There are different approaches to ordering things in source files. I know some languages prefer to put all the types or constants at the top of a file, but personally I think that's a mistake.

Source code is for humans, and the less humans have to context switch, the better off we are.  

I think files should be ordered from the main entrypoint (in this case `GenerateRoutes`) followed by all the functions and types it uses. Sometimes if the main entrypoint accepts a large `Options` struct, it might make sense to start with that, but in this case these types are implementation details of building routes, so I moved them down.

This approach allows files to be read easily from top-to-bottom, and reduces the amount of jumping around (context switching) that is required to read the source.